substrate/frame/support/src/traits/tokens/asset_ops.rs

```diff
@@ -0,0 +1,646 @@
+	impl<RuntimeOrigin, Inner: CreateStrategy> CreateStrategy for WithOrigin<RuntimeOrigin, Inner> {
+		type Success = Inner::Success;
+	}
+	impl<RuntimeOrigin, Inner: TransferStrategy> TransferStrategy for WithOrigin<RuntimeOrigin, Inner> {
+		type Success = Inner::Success;
+	}
+	impl<RuntimeOrigin, Inner: DestroyStrategy> DestroyStrategy for WithOrigin<RuntimeOrigin, Inner> {
+		type Success = Inner::Success;
+	}
+	impl<RuntimeOrigin, Inner: StashStrategy> StashStrategy for WithOrigin<RuntimeOrigin, Inner> {
+		type Success = Inner::Success;
+	}
+	impl<RuntimeOrigin, Inner: RestoreStrategy> RestoreStrategy for WithOrigin<RuntimeOrigin, Inner> {
+		type Success = Inner::Success;
+	}
+
+	/// The JustDo represents the simplest strategy,
+	/// which doesn't require additional checks to perform the operation.
+	///
+	/// It can be used as the following strategies:
+	/// * [`transfer strategy`](TransferStrategy)
+	/// * [`destroy strategy`](DestroyStrategy)
+	/// * [`stash strategy`](StashStrategy)
+	/// * [`restore strategy`](RestoreStrategy)
+	///
+	/// It accepts whatever parameters are set in its generic argument.
+	/// For instance, for an unchecked transfer,
+	/// this strategy may take a reference to a beneficiary account.
+	pub struct JustDo<Params = ()>(pub Params);
+	impl Default for JustDo<()> {
+		fn default() -> Self {
+			Self(())
+		}
+	}
+	impl<Params> TransferStrategy for JustDo<Params> {
+		type Success = ();
+	}
+	impl<Params> DestroyStrategy for JustDo<Params> {
+		type Success = ();
+	}
+	impl<Params> StashStrategy for JustDo<Params> {
+		type Success = ();
+	}
+	impl<Params> RestoreStrategy for JustDo<Params> {
+		type Success = ();
+	}
+
+	/// The `Bytes` strategy represents raw metadata bytes.
+	/// It is both an [inspect](MetadataInspectStrategy) and [update](MetadataUpdateStrategy)
+	/// metadata strategy.
+	///
+	/// * As the inspect strategy, it returns `Vec<u8>`.
+	/// * As the update strategy, it accepts `Option<&[u8]>`, where `None` means data removal.
+	///
+	/// By default, the `Bytes` identifies a byte blob associated with the asset (the only one
+	/// blob). However, a user can define several flavors of this strategy by supplying the `Flavor`
+	/// type. The `Flavor` type can also contain additional data (like a byte key) to identify a
+	/// certain byte data.
+	pub struct Bytes<Flavor = ()>(pub Flavor);
+	impl Default for Bytes<()> {
+		fn default() -> Self {
+			Self(())
+		}
+	}
+	impl<Flavor> MetadataInspectStrategy for Bytes<Flavor> {
+		type Value = Vec<u8>;
+	}
+	impl<Flavor> MetadataUpdateStrategy for Bytes<Flavor> {
+		type Update<'u> = Option<&'u [u8]>;
+		type Success = ();
+	}
+
+	/// The `Ownership` [inspect](MetadataInspectStrategy) metadata strategy allows getting the
+	/// owner of an asset.
+	pub struct Ownership<Owner>(PhantomData<Owner>);
+	impl<Owner> Default for Ownership<Owner> {
+		fn default() -> Self {
+			Self(PhantomData)
+		}
+	}
+	impl<Owner> MetadataInspectStrategy for Ownership<Owner> {
```

Comment:
I see what you mean by "strategy" here, it's more of a view inside the metadata. You specify some values that are reasonable for the metadata to have, and a catch-all `bytes` one for custom stuff. It's maybe more of a `Field`?

Reply:
I agree. It indeed feels like a `Field` or something.